server/eclipse-project/src/main/java/us/freeandfair/corla/json/DoSDashboardRefreshResponse.java

```diff
@@ -120,22 +135,45 @@ protected DoSDashboardRefreshResponse(final ASMState the_asm_state,
    * @exception NullPointerException if necessary information to construct the
    * response does not exist.
    */
+  @SuppressWarnings("checkstyle:magicnumber")
   public static DoSDashboardRefreshResponse 
       createResponse(final DoSDashboard the_dashboard) {
     // construct the various audit info from the contests to audit in the dashboard
-    final Map<Long, AuditReason> audited_contests = 
-        new HashMap<Long, AuditReason>();
-    final Map<Long, Integer> estimated_ballots_to_audit = new HashMap<Long, Integer>();
-    final Map<Long, Integer> optimistic_ballots_to_audit = new HashMap<Long, Integer>();
-    final Set<Long> hand_count_contests = new HashSet<Long>();
+    final SortedMap<Long, AuditReason> audited_contests = 
+        new TreeMap<Long, AuditReason>();
+    final SortedMap<Long, Integer> estimated_ballots_to_audit = new TreeMap<Long, Integer>();
+    final SortedMap<Long, Integer> optimistic_ballots_to_audit = new TreeMap<Long, Integer>();
+    final SortedMap<Long, Map<Integer, Integer>> discrepancy_count = new TreeMap<>();
+    final List<Long> hand_count_contests = new ArrayList<Long>();
     
     for (final ContestToAudit cta : the_dashboard.contestsToAudit()) {
       switch (cta.audit()) {
         case COMPARISON:
           audited_contests.put(cta.contest().id(), cta.reason());
-          final Pair<Integer, Integer> estimates = ballotsToAudit(cta.contest());
-          estimated_ballots_to_audit.put(cta.contest().id(), estimates.first());
-          optimistic_ballots_to_audit.put(cta.contest().id(), estimates.second());
+          final Map<Integer, Integer> discrepancy = new HashMap<>();
+          int optimistic = Integer.MIN_VALUE;
+          int estimated = Integer.MIN_VALUE;
+          for (final CountyContestComparisonAudit ccca : 
+               CountyContestComparisonAuditQueries.matching(cta.contest())) {
+            optimistic = 
+                Math.max(optimistic, 
+                         Math.max(0, ccca.optimisticBallotsToAudit() - 
+                                     ccca.dashboard().auditedPrefixLength()));
+            estimated = 
+                Math.max(estimated, 
+                         Math.max(0, ccca.estimatedBallotsToAudit() - 
+                                     ccca.dashboard().auditedPrefixLength()));
+            for (int i = -2; i <= 2; i++) {
```

Comment:
Maybe add a comment here to note that these are the possible discrepancy values. It was obvious after recent code review, but since it isn't encoded in an enum, might be worth remarking.

Reply:
I have been meaning to encode it in an enum. Maybe now is the right time to do that.

Reply:
Or, maybe after this build is delivered to CDOS and I have a bit more time to do the right thing...